### PR TITLE
🐛 fix display of current/set temperature when 0

### LINKED ISSUE #4366: Thermostat card shows nothing when setpoint is set to 0.0
**Checklist:**

- [X] I updated to the latest version available
- [X] I cleared the cache of my browser

**Home Assistant release with the issue:**
0.103.0

**Last working Home Assistant release (if known):**
Not known

**UI (States or Lovelace UI?):**
Lovelace

**Browser and Operating System:**
Firefox 71.0 on Windows 7

**Description of problem:**
As the title says, the setpoint value is not shown when it is set to 0.0.

Setpoint at 0.5:
image

Setpoint at 0.0:
image

**Expected behaviour:**
Shows 0.0

**Relevant config:**

```
  - platform: generic_thermostat
    name: Câble chauffant
    heater: switch.puits
    target_sensor: sensor.puits_ds18b20_2_temperature
    min_temp: -5
    max_temp: 15
    ac_mode: False
    cold_tolerance: 0.5
    hot_tolerance: 0.0
    min_cycle_duration:
      minutes: 2
    keep_alive:
      minutes: 1
    precision: 0.1
```

**Steps to reproduce this problem:**

1. Set the desired temperature to 0.0

### PR DESCRIPTION
**Pull request by iantrich**
fixes https://github.com/home-assistant/home-assistant-polymer/issues/4366

### CODE REVIEW

File: src/panels/lovelace/cards/hui-thermostat-card.ts [hunk 1]
```diff
@@ -129,7 +129,7 @@ export class HuiThermostatCard extends LitElement implements LovelaceCard {
             ></round-slider>
           `;
 
-    const currentTemperature = stateObj.attributes.current_temperature
+    const currentTemperature = !isNaN(stateObj.attributes.current_temperature)
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-thermostat-card.ts]: `currentTemperature` is a boolean now?

File: src/panels/lovelace/cards/hui-thermostat-card.ts [hunk 2]
```diff
@@ -153,7 +153,7 @@ export class HuiThermostatCard extends LitElement implements LovelaceCard {
         <g>
           <text text-anchor="middle" class="set-value">
             ${
-              !this._setTemp
+              this._setTemp === undefined
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-thermostat-card.ts]:
```suggestion
              this._setTemp === undefined || this._setTemp === null
```
Better ---> 
<https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#nullish-coalescing>

File: src/panels/lovelace/cards/hui-thermostat-card.ts (hunk 1 above)
Comment (iantrich) [reply src/panels/lovelace/cards/hui-thermostat-card.ts]: No, it is a svg or nothing, this is the check

Comment (bramkragten) [reply src/panels/lovelace/cards/hui-thermostat-card.ts]: https://github.com/home-assistant/home-assistant-polymer/blob/39f72da115a9ff0a55796971d878f730e3cd792a/src/panels/lovelace/cards/hui-thermostat-card.ts#L230

**Review by bramkragten — APPROVED** [review]

**Pull request merged**